buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016-2024 JetBrains s.r.o.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+import java.util.*
+import java.io.FileInputStream
+
+plugins {
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+    if (project.hasProperty("kotlin_repo_url")) {
```

Comment:
Use the nullable `findProperty` method to check and retrieve property. The `properties` property performs a copy of the underlying map.

Reply:
Good catch, although this block wasn't supposed to make it in at all; I'll drop